### PR TITLE
refactor(function): Update Dependency Function

### PR DESCRIPTION
**Pull request by souravdasslg**
1. Create Interface UpdateDependencyConfig
2. Update Functions
3. Update Tets

#4996

### CODE REVIEW

**Comment by rarkins** [comment]
Unfortunately I think the new naming is making it less intuitive for one-time contributors. e.g.

image

and

image

I would much prefer that the param is known as "upgrade" still.

File: lib/manager/common.ts [hunk 1]
```diff
@@ -173,6 +174,11 @@ export interface UpdateArtifactsResult {
   file?: { name: string; contents: string };
 }
 
+export interface UpdateDependencyConfig {
+  fileContent: string;
+  updateOptions: Upgrade;
```
Comment (rarkins) [inline lib/manager/common.ts]: Is there anything stopping this from being `upgrade`?

Comment (souravdasslg) [reply lib/manager/common.ts]: Initially, it was a hard choice for me too. I realized after analyzing the complete functionality, and it is not necessarily upgrading until conditions are met.
So, all we are passing here are a couple of data points to the function to make a decision and make changes based on those conditions.

Comment (rarkins) [reply lib/manager/common.ts]: Internally we refer to these as “upgrades” so it’s consistent that this param is called upgrade. Unless I’m forgetting something, it’s very rare that an upgrade would not lead to a branch/PR. Sometimes there’s “no work” to do, but that only means “upgrade was already applied” and not “this is not an upgrade”

File: lib/manager/common.ts [hunk 2]
```diff
@@ -161,6 +161,7 @@ export interface Upgrade<T = Record<string, any>>
   toVersion?: string;
   updateType?: string;
   version?: string;
+  nestedVersion?: boolean;
```
Comment (rarkins) [inline lib/manager/common.ts]: Where does the need for this come from? I thought it should always be a part of `managerData`

Comment (souravdasslg) [reply lib/manager/common.ts]: Let me check this for the exact details. As I remember, only one function was passing `nestedVersion` value, and I added it to the interface.

**Review by rarkins — CHANGES_REQUESTED** [review]
Check `nestedVersion`

**Review by viceice — CHANGES_REQUESTED** [review]
needs deconflicting

**Review by rarkins — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**